### PR TITLE
Xqiu 20170818 add reconciliation

### PR DESCRIPTION
**Pull request by unieagle**
This PR is for https://github.com/killbill/killbill-cybersource-plugin/issues/19
add optional reconciliation_id, if present, will pass to Cybersource and record it even if the transaction failed.
@pierre please review, thanks!

### CODE REVIEW

File: lib/cybersource/models/response.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def self.cybersource_response_params(response)
             :params_cv_code => extract(response, 'cvCode'),
             :params_authorized_date_time => extract(response, 'authorizedDateTime'),
             :params_processor_response => extract(response, 'processorResponse'),
-            :params_reconciliation_id => extract(response, 'reconciliationID'),
+            :params_reconciliation_id => extract(response, 'reconciliationID') || extract(response, 'reconciliation_id'),
```
Comment (pierre) [inline lib/cybersource/models/response.rb]: Do we need this? The `response` comes from CyberSource, so it should be camel-cased.

Comment (unieagle) [inline lib/cybersource/models/response.rb]: the `reconciliationID` comes from Cybersource response, but in the case of transaction failures, there won't be this value. So use `reconciliation_id`, which is the value received in the request. see this line: https://github.com/killbill/killbill-cybersource-plugin/pull/20/files#diff-ac9d5b9b8d335656dab060b783053300R178

Comment (pierre) [inline lib/cybersource/models/response.rb]: Oh I see - can you add a comment? Otherwise, I'm afraid we might _clean_ it up one day...

**Comment by unieagle** [comment]
@pierre , comment added.

**Comment by sbrossie** [comment]
🎉 Well done @unieagle !

**Pull request merged**